src/ray/gcs/redis_job_info_accessor.h

```diff
@@ -0,0 +1,53 @@
+#ifndef RAY_GCS_REDIS_JOB_INFO_ACCESSOR_H
+#define RAY_GCS_REDIS_JOB_INFO_ACCESSOR_H
+
+#include "ray/common/id.h"
+#include "ray/gcs/callback.h"
+#include "ray/gcs/job_info_accessor.h"
+#include "ray/gcs/subscription_executor.h"
+#include "ray/gcs/tables.h"
+
+namespace ray {
+
+namespace gcs {
+
+class RedisGcsClient;
+
+/// \class RedisJobInfoAccessor
+/// RedisJobInfoAccessor is an implementation of `JobInfoAccessor`
+/// that uses Redis as the backend storage.
+class RedisJobInfoAccessor : public JobInfoAccessor {
+ public:
+  explicit RedisJobInfoAccessor(RedisGcsClient *client_impl);
```

Comment:
Avoid raw pointer here.

Reply:
We can add a CHECK here if your concern is the pointer is not initialized.

Reply:
I see, in this case can we change `client_impl_` to a reference to indicate non-nullability?